server/src/main/java/org/elasticsearch/action/bulk/BulkOperation.java

```diff
@@ -298,6 +298,10 @@ private void executeBulkRequestsByShard(
                     bulkRequest.getRefreshPolicy(),
                     requests.toArray(new BulkItemRequest[0])
                 );
+                var indexMetadata = clusterState.getMetadata().index(shardId.getIndexName());
+                if (indexMetadata != null && indexMetadata.getInferenceFields().isEmpty() == false) {
+                    bulkShardRequest.setInferenceFieldMap(indexMetadata.getInferenceFields());
```

Comment:
Add inference metadata to bulk shard request so it can be retrieved by the `ActionFilter`

Reply:
I commented on the bulk shard request object, but not serializing these makes me nervous and we need to ensure they are null when writing to the wire. We should throw an internal server exception in those cases as we have a bug.

Reply:
Changed in 4240f6b38c0c56e5d90505af01f299e3b3277438

Reply:
Added check on serialization in 918a695f5b7c17a85f4ef3d28839e54e4f231a59